pkg/client/client_test.go

```diff
@@ -164,6 +185,40 @@ func TestApplySingle(t *testing.T) {
 	})
 }
 
+func newClient(t *testing.T, s *runtime.Scheme) (*applyCl.ApplyClient, *test.FakeClient) {
+	cli := NewFakeClient(t)
+	cli.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+		// force the Generation to `1` for newly created objects
+		if obj, ok := obj.(*unstructured.Unstructured); ok {
```

Comment:
This won't work - you cannot cast to `unstructured.Unstructured` - it will always return false.
you should use the `meta.Accessor` instead:
```go
ob, err := meta.Accessor(obj)
if err != nil {
	return err
}
ob.SetGeneration(1)
```

Reply:
ah thanks! and yes, I saw it failed, but I thought I used something else instead 🤦‍♂

Reply:
fixed in cd909f5